src/app/exam/problems/ExamProblems.js

```diff
@@ -0,0 +1,171 @@
+import { ThreeDotsButton } from "../../commons/buttons/ThreeDotsButton.js";
+import FakeLink from "../../commons/FakeLink.js";
+import { ItemListPage } from "../../commons/ItemListPage/ItemListPage.js";
+import { ExamInPageNavigationBar } from "../ExamInPageNavigationBar";
+import { AddProblemModal } from "../modals/AddProblemModal.js";
+import { EditProblemModal } from "../modals/EditProblemModal.js";
+import {RejudgeProblemModal} from "../modals/RejudgeProblemModal";
+import "./ExamProblems.scss";
+
+const toCharacterIndex = i => {
+    return String.fromCharCode(i + 65);
+}
+
+
+const ExamProblems = ({ exams }) => {
+    const { url: currentURL } = useRouteMatch()
+    const { examId } = useParams()
+    const currentExamName = exams.find(exam => exam.id === parseInt(examId))?.name
+    const [examProblems, setExamProblems] = useState(null);
+
+    const [showAddProblemModal, setShowAddProblemModal] = useState(false);
+    const [showRejudgeProblemModal, setShowRejudgeProblemModal] = useState(0);
+    const [rejudgeProblemId, setRejudgeProblemId] = useState(0)
+    const [showEditProblemModal, setShowEditProblemModal] = useState(false);
+
+    const [editingProblem, setEditingProblem] = useState(null);
+
+    const fetchExam = useCallback(() => {
+        examService.getExamOverview(examId).then(exam => {
+            exam.questions.sort((questionA, questionB) => questionA.questionOrder - questionB.questionOrder);
+            setExamProblems(exam.questions);
+        });
+    }, [examId])
+
+    useEffect(() => {
+        if (!examProblems) {
+            fetchExam()
+        }
+    }, [examProblems, fetchExam]);
+
+    const editProblem = (problemId) => {
+        const editQuestionPromise = examService.editExamQuestion({examId, problemId});
+        editQuestionPromise.then(fetchExam);
+        return editQuestionPromise;
+    };
+
+    const dropDownItems = (examProblem) => [{
+        name: "Edit",
+        dangerous: false,
+        onClick: () => {
+            setEditingProblem(examProblem)
+            setShowEditProblemModal(true)
+        }
+    }, {
+        name: "Rejudge",
+        dangerous: false,
+        onClick: () => {
+            setShowRejudgeProblemModal(examProblem.problemId)
+        }
+    }, {
+        name: "Delete",
+        dangerous: true,
+        onClick: () => {
+            examService.deleteExamProblem({
+                examId: examId,
+                problemId: examProblem
+            }).then(res => console.log("deleted", res))
+        }
+    }];
+
+    const rejudgeProblem = (problemId) => {
+        setShowRejudgeProblemModal(0)
+        setRejudgeProblemId(problemId)
+        submissionService.rejudge({
+            examId: examId,
+            problemId: problemId
+        }).then(res => {
+            console.log(res)
+        })
+        setTimeout(() =>
+        {
+            setRejudgeProblemId(0)
+        }, 5000)
```

Comment:
why is the need for `setTimeout`? 

do this in the "then" section 

and what does `0` mean?